src/libraries/System.Runtime.InteropServices/gen/ComInterfaceGenerator/ComInterfaceGenerator.cs

```diff
@@ -185,6 +185,18 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
                     using StringWriter source = new();
                     source.WriteLine("// <auto-generated />");
                     source.WriteLine("#pragma warning disable CS0612, CS0618"); // Suppress warnings about [Obsolete] member usage in generated code.
+
+                    // If the user has specified 'ManagedObjectWrapper', it means that the COM interface will never be used to marshal a native
+                    // object as an RCW (eg. the IDIC vtable will also not be generated, nor any additional supporting code). To reduce binary
+                    // size, we're not emitting the interface methods on the implementation interface that has '[DynamicInterfaceCastableImplementation]'
+                    // on it. However, doing so will cause the CA2256 warning to be produced. We can't remove the attribute, as that would cause
+                    // the wrong exception to be thrown when trying an IDIC cast with this interface (not 'InvalidCastException'). Because this is
+                    // a niche scenario, and we don't want to regress perf or size, we can just disable the warning instead.
+                    if (interfaceContext.Options is ComInterfaceOptions.ManagedObjectWrapper)
+                    {
+                        source.WriteLine("#pragma warning disable CA2256");
```

Comment:
There should also be a close warning to reenable the warning it after the type definition.

Reply:
Just making sure I understand, why? Don't we also just keep the other CS06XX warnings suppressed for the whole file? My only concern is that reenabling it after the interface definition might be a more complicated change 😅

Reply:
> Just making sure I understand, why?

We typically narrow our disabling of warnings to the smallest scope when in file. I don't see anything in the `docs/` about this, but that is the general pattern in the BCL and VM and just good practice for warning suppression in my opinion.

@stephentoub Opinions?

> Don't we also just keep the other CS06XX warnings suppressed for the whole file? 

Suppressing warnings broadly is generally bad practice, but we can discuss the policy elsewhere if we're already doing this here.

Reply:
> "Please add a new line suppressing this warning throughout the file."

I'm... Not sure I'm following, isn't the code in this PR already doing that? 😅
I did add:
```csharp
source.WriteLine("#pragma warning disable CA2256");
```

Reply:
In the same location as the other is what I'm suggesting.

Reply:
But it is, it's right after that one 😅
Are you saying you always want to emit this suppression, even when not needed?

Reply:
No, this is fine. I misread where this was. Thanks.